src/hooks/useAppUpdate.ts

```diff
@@ -35,31 +32,16 @@ export const useAppUpdate = () => {
     isDev && overriddenAppVersion ? overriddenAppVersion : getVersion();
 
   // Parse the update text JSON for internationalization
-  const getUpdateMessage = useCallback(() => {
-    if (!updateText.enabled || !updateText.payload) {
-      return t("appUpdate.defaultMessage");
-    }
-
-    if (typeof updateText.payload === "object" && updateText.payload !== null) {
-      const currentLanguage = i18n.language;
-      const payload = updateText.payload as Record<string, string>;
-      return (
-        payload[currentLanguage] || payload.en || t("appUpdate.defaultMessage")
-      );
-    }
-
-    return t("appUpdate.defaultMessage");
-  }, [updateText.enabled, updateText.payload, i18n.language, t]);
-
-  const updateMessage = getUpdateMessage();
+  const updateMessage = getAppUpdateText(updateText);
 
   // Only check for updates when remote config is initialized
+  // For this app, any version difference (including protocol) requires a forced update
```

Comment:
let's remove this comment (and changes) and use the flags to decide which app update should be "required" or "optional" like the PR was originally doing - thanks!